Updating bq config doc

nataliefiann opened the pull request:
## What are you changing in this pull request and why?
<!---
Describe your changes and why you're making them. If related to an open 
issue or a pull request on dbt Core, then link to them here! 

To learn more about the writing conventions used in the dbt Labs docs, see the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md).
-->

I'm updating this document following a message / request from a Support representative where it was advised the speciying tag section of the doc could clearer and more specific: https://dbt-labs.slack.com/archives/C067EUS2D7S/p1720529616317879

## Checklist
<!--
Uncomment when publishing docs for a prerelease version of dbt:
- [ ] Add versioning components, as described in [Versioning Docs](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-entire-pages)
- [ ] Add a note to the prerelease version [Migration Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/docs/dbt-versions/core-upgrade)
-->
- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [ ] For [docs versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#about-versioning), review how to [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content).
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding or removing pages (delete if not applicable):
- [ ] Add/remove page in `website/sidebars.js`
- [ ] Provide a unique filename for new pages
- [ ] Add an entry for deleted pages in `website/vercel.json`
- [ ] Run link testing locally with `npm run build` to update the links that point to deleted pages

nataliefiann committed:
Updating bq config doc

mirnawong1 commented on website/docs/reference/resource-configs/bigquery-configs.md:
```diff
@@ -378,7 +378,7 @@ models:
 
 ### Specifying tags
-BigQuery table and view *tags* can be created by supplying an empty string for the label value.
+This isn't the tag you're looking for (it's just a label with no value).
```
i think we may want to keep this info the same - I've asked in the slack thread to confirm

nataliefiann replied:
Thanks Mirna

mirnawong1 commented on website/docs/reference/resource-configs/bigquery-configs.md:
```diff
@@ -395,6 +395,11 @@ select * from {{ ref('another_model') }}
 
 </File>
 
+Sometimes a label with key with an empty value is referred to a tag. However, this should not be confused with a [tag resource](/bigquery/docs/tags). You can find out more in [labels and tags](/resource-manager/docs/tags/tags-overview). You can create a new label with no value, or you can remove a value from an existing label key.
```
if this content is the same as the google one, we can just provide a concise sentence and link to the google doc so it's less maintenance should the google docs change.

mirnawong1 replied:
```suggestion
Sometimes a label with key with an empty value is referred to a tag. However, this should not be confused with a [tag resource](https://cloud.google.com/bigquery/docs/tags). You can find out more in [labels and tags](/resource-manager/docs/tags/tags-overview). You can create a new label with no value, or you can remove a value from an existing label key.
```

nataliefiann replied:
```suggestion
A label with an empty value is also called a tag, but this shouldn't be confused with with a tag resource which is suitable for setting iam permissions. You can find out more about this in [Tag tables and databases](https://cloud.google.com/bigquery/docs/tags).
```

mirnawong1 committed:
Merge branch 'current' into nfiann-bq-config

nataliefiann commented on website/docs/reference/resource-configs/bigquery-configs.md:
```diff
@@ -378,7 +378,7 @@ models:
 
 ### Specifying tags
-BigQuery table and view *tags* can be created by supplying an empty string for the label value.
+This isn't the tag you're looking for (it's just a label with no value).
```
```suggestion
BigQuery table and view *tags* can be created by supplying an empty string for the label value..
```

nataliefiann committed:
Update website/docs/reference/resource-configs/bigquery-configs.md

nataliefiann commented on website/docs/reference/resource-configs/bigquery-configs.md:
```diff
@@ -378,7 +378,7 @@ models:
 
 ### Specifying tags
-BigQuery table and view *tags* can be created by supplying an empty string for the label value.
+BigQuery table and view *tags* can be created by supplying an empty string for the label value..
```
```suggestion
BigQuery table and view *tags* can be created by supplying an empty string for the label value.
```

nataliefiann committed:
Update website/docs/reference/resource-configs/bigquery-configs.md

nataliefiann committed:
Update website/docs/reference/resource-configs/bigquery-configs.md

A label with an empty value is also called a tag, but this shouldn't be confused with with a tag resource, which is suitable for setting iam permissions. You can read more about this in [tag resource](https://cloud.google.com/bigquery/docs/tags)

Co-authored-by: Mirna Wong <89008547+mirnawong1@users.noreply.github.com>

nataliefiann commented on website/docs/reference/resource-configs/bigquery-configs.md:
```diff
@@ -395,6 +395,11 @@ select * from {{ ref('another_model') }}
 
+Sometimes a label with key with an empty value is referred to a tag. However, this should not be confused with a [tag resource](/bigquery/docs/tags). You can find out more in [labels and tags](/resource-manager/docs/tags/tags-overview). You can create a new label with no value, or you can remove a value from an existing label key.
+
+In cases where you are labeling a resource, using labels without values can be beneficial, and as a note, you won't need the key-value format. For example, if a table contains test data that is used by multiple groups, such as support or development, you can add a `test_data` label to the table to identify it.
```
```suggestion
For more information, you can also look at [Tags overview](https://cloud.google.com/resource-manager/docs/tags/tags-overview)
```

nataliefiann committed:
Update website/docs/reference/resource-configs/bigquery-configs.md

nataliefiann commented on website/docs/reference/resource-configs/bigquery-configs.md:
```diff
@@ -395,6 +395,11 @@ select * from {{ ref('another_model') }}
+Sometimes a label with key with an empty value is referred to a tag. However, this should not be confused with a [tag resource](https://cloud.google.com/bigquery/docs/tags). You can find out more in [labels and tags](/resource-manager/docs/tags/tags-overview). You can create a new label with no value, or you can remove a value from an existing label key.
+
+For more information, you can also look at [Tags overview](https://cloud.google.com/resource-manager/docs/tags/tags-overview)
+
```
```suggestion
It's not currently possible to apply IAM tags in Big Query. However, we have an open [feature request](https://github.com/dbt-labs/dbt-bigquery/issues/1134) which you can view and upvote in GitHub.
```

nataliefiann committed:
Update website/docs/reference/resource-configs/bigquery-configs.md

mirnawong1 committed:
Merge branch 'current' into nfiann-bq-config

mirnawong1 commented on website/docs/reference/resource-configs/bigquery-configs.md:
```diff
@@ -395,6 +395,11 @@ select * from {{ ref('another_model') }}
 
 </File>
 
+Sometimes a label with key with an empty value is referred to a tag. However, this should not be confused with a [tag resource](https://cloud.google.com/bigquery/docs/tags). You can find out more in [labels and tags](/resource-manager/docs/tags/tags-overview). You can create a new label with no value, or you can remove a value from an existing label key.
```
```suggestion
Sometimes a label with key with an empty value is referred to a tag. However, this should not be confused with a [tag resource](https://cloud.google.com/bigquery/docs/tags). You can find out more in [labels and tags](https://cloud.google.com/resource-manager/docs/tags/tags-overview). You can create a new label with no value, or you can remove a value from an existing label key.
```

nataliefiann committed:
Update website/docs/reference/resource-configs/bigquery-configs.md

Co-authored-by: Mirna Wong <89008547+mirnawong1@users.noreply.github.com>

mirnawong1 commented on website/docs/reference/resource-configs/bigquery-configs.md:
```diff
@@ -395,6 +395,11 @@ select * from {{ ref('another_model') }}
 
 </File>
 
+Sometimes a label with key with an empty value is referred to a tag. However, this should not be confused with a [tag resource](https://cloud.google.com/bigquery/docs/tags). You can find out more in [labels and tags](https://cloud.google.com/resource-manager/docs/tags/tags-overview). You can create a new label with no value, or you can remove a value from an existing label key.
```
i feel like the last sentence fits better at the beginning since it follows on from the previous sentence about supplying an empty string for the label value.

also added specific info what tag resource means (iam)

```suggestion
You can create a new label with no value or remove a value from an existing label key.

Sometimes a label with a key that has an empty value is referred to as a tag. However, this should not be confused with a [tag resource](https://cloud.google.com/bigquery/docs/tags), which conditionally applies IAM policies to BigQuery tables and datasets. Find out more in [labels and tags](https://cloud.google.com/resource-manager/docs/tags/tags-overview). 
```

mirnawong1 commented on website/docs/reference/resource-configs/bigquery-configs.md:
```diff
@@ -395,6 +395,11 @@ select * from {{ ref('another_model') }}
 
+Sometimes a label with key with an empty value is referred to a tag. However, this should not be confused with a [tag resource](https://cloud.google.com/bigquery/docs/tags). You can find out more in [labels and tags](https://cloud.google.com/resource-manager/docs/tags/tags-overview). You can create a new label with no value, or you can remove a value from an existing label key.
+
+For more information, you can also look at [Tags overview](https://cloud.google.com/resource-manager/docs/tags/tags-overview)
```
we're linking to the overview doc twice, so not sure if it's needed?
```suggestion
```

mirnawong1 commented on website/docs/reference/resource-configs/bigquery-configs.md:
```diff
@@ -395,6 +395,11 @@ select * from {{ ref('another_model') }}
+Sometimes a label with key with an empty value is referred to a tag. However, this should not be confused with a [tag resource](https://cloud.google.com/bigquery/docs/tags). You can find out more in [labels and tags](https://cloud.google.com/resource-manager/docs/tags/tags-overview). You can create a new label with no value, or you can remove a value from an existing label key.
+
+For more information, you can also look at [Tags overview](https://cloud.google.com/resource-manager/docs/tags/tags-overview)
+It's not currently possible to apply IAM tags in Big Query. However, we have an open [feature request](https://github.com/dbt-labs/dbt-bigquery/issues/1134) which you can view and upvote in GitHub.
```
```suggestion
Currently, it's not possible to apply IAM tags in BigQuery, however, you can contribute by upvoting [this issue](https://github.com/dbt-labs/dbt-bigquery/issues/1134) on GitHub.
```

mirnawong1 reviewed (changes requested):
hey @nataliefiann, great job simplifying this! left some suggestions

nataliefiann committed:
Update website/docs/reference/resource-configs/bigquery-configs.md

Co-authored-by: Mirna Wong <89008547+mirnawong1@users.noreply.github.com>

nataliefiann committed:
Update website/docs/reference/resource-configs/bigquery-configs.md

Co-authored-by: Mirna Wong <89008547+mirnawong1@users.noreply.github.com>

nataliefiann committed:
Update website/docs/reference/resource-configs/bigquery-configs.md

Co-authored-by: Mirna Wong <89008547+mirnawong1@users.noreply.github.com>

nataliefiann commented:
Hiya @mirnawong1 

Thank you providing your feedback and suggestions. It's really helpful and neatens up the doc. I'll take these comments on board going forward.

Kind Regards

nataliefiann commented on website/docs/reference/resource-configs/bigquery-configs.md:
```diff
@@ -395,6 +395,10 @@ select * from {{ ref('another_model') }}
+Sometimes a label with a key that has an empty value is referred to as a tag. However, this should not be confused with a [tag resource](https://cloud.google.com/bigquery/docs/tags). You can find out more in [labels and tags](https://cloud.google.com/resource-manager/docs/tags/tags-overview). You can create a new label with no value, or you can remove a value from an existing label key.
+
+Currently, it's not possible to apply IAM tags in BigQuery, however, you can contribute by upvoting [this issue](https://github.com/dbt-labs/dbt-bigquery/issues/1134) on GitHub.
+
```
```suggestion
You can create a new label with no value or remove a value from an existing label key.
Sometimes a label with a key that has an empty value is referred to as a tag. However, this should not be confused with a [tag resource](https://cloud.google.com/bigquery/docs/tags), which conditionally applies IAM policies to BigQuery tables and datasets. Find out more in [labels and tags](https://cloud.google.com/resource-manager/docs/tags/tags-overview). 
```

nataliefiann committed:
Update website/docs/reference/resource-configs/bigquery-configs.md

nataliefiann commented on website/docs/reference/resource-configs/bigquery-configs.md:
```diff
@@ -395,6 +395,11 @@ select * from {{ ref('another_model') }}
 
 </File>
 
+Sometimes a label with a key that has an empty value is referred to as a tag. However, this should not be confused with a [tag resource](https://cloud.google.com/bigquery/docs/tags). You can find out more in [labels and tags](https://cloud.google.com/resource-manager/docs/tags/tags-overview). You can create a new label with no value, or you can remove a value from an existing label key.
```
```suggestion

```

nataliefiann commented on website/docs/reference/resource-configs/bigquery-configs.md:
```diff
@@ -395,6 +395,11 @@ select * from {{ ref('another_model') }}
+
+Currently, it's not possible to apply IAM tags in BigQuery, however, you can contribute by upvoting [this issue](https://github.com/dbt-labs/dbt-bigquery/issues/1134) on GitHub.
+You can create a new label with no value or remove a value from an existing label key.
+Sometimes a label with a key that has an empty value is referred to as a tag. However, this should not be confused with a [tag resource](https://cloud.google.com/bigquery/docs/tags), which conditionally applies IAM policies to BigQuery tables and datasets. Find out more in [labels and tags](https://cloud.google.com/resource-manager/docs/tags/tags-overview). 
```
```suggestion
You can create a new label with no value or remove a value from an existing label key.

Sometimes a label with a key that has an empty value is referred to as a tag. However, this should not be confused with a [tag resource](https://cloud.google.com/bigquery/docs/tags), which conditionally applies IAM policies to BigQuery tables and datasets. Find out more in [labels and tags](https://cloud.google.com/resource-manager/docs/tags/tags-overview). 

Currently, it's not possible to apply IAM tags in BigQuery, however, you can contribute by upvoting [this issue](https://github.com/dbt-labs/dbt-bigquery/issues/1134) on GitHub.

```

nataliefiann committed:
Update website/docs/reference/resource-configs/bigquery-configs.md

nataliefiann committed:
Merge branch 'current' into nfiann-bq-config

nataliefiann committed:
Update website/docs/reference/resource-configs/bigquery-configs.md

nataliefiann committed:
Update bigquery-configs.md

Deleted repeated sentences

mirnawong1 commented on website/docs/reference/resource-configs/bigquery-configs.md:
```diff
@@ -395,6 +395,10 @@ select * from {{ ref('another_model') }}
 
 </File>
 
+You can create a new label with no value or remove a value from an existing label key. A label with a key that has an empty value can also be referred to as a tag. However, this should not be confused with a [tag resource](https://cloud.google.com/bigquery/docs/tags), which conditionally applies IAM policies to BigQuery tables and datasets. Find out more in [labels and tags](https://cloud.google. ...
```
sugg adding a line break and adding the clarification that bigquery uses tag for both things
```suggestion
You can create a new label with no value or remove a value from an existing label key. 

A label with a key that has an empty value can also be [referred](https://cloud.google.com/bigquery/docs/adding-labels#adding_a_label_without_a_value) to as a tag in BigQuery. However, this should not be confused with a [tag resource](https://cloud.google.com/bigquery/docs/tags), which conditionally applies IAM policies to BigQuery tables and datasets. Find out more in [labels and tags](https://cloud.google.com/resource-manager/docs/tags/tags-overview). 
```

mirnawong1 commented on website/docs/reference/resource-configs/bigquery-configs.md:
```diff
@@ -395,6 +395,10 @@ select * from {{ ref('another_model') }}
 
+You can create a new label with no value or remove a value from an existing label key. A label with a key that has an empty value can also be referred to as a tag. However, this should not be confused with a [tag resource](https://cloud.google.com/bigquery/docs/tags), which conditionally applies IAM policies to BigQuery tables and datasets. Find out more in [labels and tags](https://cloud.google. ...
+
+Currently, it's not possible to apply IAM tags in BigQuery, however, you can contribute by upvoting [this issue](https://github.com/dbt-labs/dbt-bigquery/issues/1134) on GitHub.
```
```suggestion
Currently, it's not possible to apply IAM tags in BigQuery, however, you can weigh in by upvoting [GitHub issue 1134](https://github.com/dbt-labs/dbt-bigquery/issues/1134).
```

mirnawong1 reviewed (approved):
looks great! i have some minor non blocker suggestions so will approve this for you!

am going to cc @jtcohen6 in case there's add'l technical feedback though!

mirnawong1 committed:
Merge branch 'current' into nfiann-bq-config

mirnawong1 committed:
Merge branch 'current' into nfiann-bq-config

amychen1776 committed:
Merge branch 'current' into nfiann-bq-config

amychen1776 committed:
Update website/docs/reference/resource-configs/bigquery-configs.md

Co-authored-by: Mirna Wong <89008547+mirnawong1@users.noreply.github.com>

amychen1776 committed:
Update website/docs/reference/resource-configs/bigquery-configs.md

Co-authored-by: Mirna Wong <89008547+mirnawong1@users.noreply.github.com>
